### PR TITLE
DEV: Move hotlinked image information into a dedicated table

### PR DESCRIPTION
**Pull request by davidtaylorhq**
This will make future changes to the 'pull hotlinked images' system easier. This commit should not introduce any functional change.

For now, the old post_custom_field data is kept in the database. This will be dropped in a future commit.

### CODE REVIEW

**Comment by CvX** [comment]
(backend specs say "no" 😛)

File: app/models/post.rb [hunk 1]
```diff
@@ -78,6 +80,8 @@ class Post < ActiveRecord::Base
 
   SHORT_POST_CHARS ||= 1200
 
+  # TODO: Drop the data from these three custom fields,
```
Comment (pmusaraj) [inline app/models/post.rb:83]: Can we add a "after n" version to this todo comment?

File: spec/lib/cooked_post_processor_spec.rb [hunk 2]
```diff
@@ -1137,6 +1135,7 @@
 
       post = Fabricate(:post, raw: url)
 
+      PostHotlinkedMedia.create!(url: "//image.com/avatar.png", post: post, status: 'too_large')
```
Comment (pmusaraj) [inline spec/lib/cooked_post_processor_spec.rb:1138]: Is it possible to add a test case for broken images?

**Review by pmusaraj — APPROVED** [review]
Looks great, minor comments only.

File: app/models/post.rb (hunk 1 above)
Comment (davidtaylorhq) [reply app/models/post.rb:83]: I intend to drop the old data within a day or two of merging (once we're 100% confident the change won't need to be reverted). Just opened a draft PR here so we don't forget: #16594

**Pull request merged**